### ISSUE TITLE
Easy to use dataset for DALL-E

### ISSUE THREAD

**Issue by mrconter1** [lucidrains/DALLE-pytorch#15]
The [COCO dataset](https://cocodataset.org/#overview) is a high-quality dataset both in terms of images and text. Each image has multiple captions. I have created a [Google Colab](https://colab.research.google.com/drive/1KxG1iGBoKt2fLVH7uXG_vhvll2OlFkey?usp=sharing) which does the following:

1. Fetches the COCO images and captions
2. Allows the user to specify image dimensions which the images will be resized to
3. Stores information in two files for easy access

_od-captions.txt_ (<image_path> : <image_caption>)
> train2017/000000203564.jpg : A bicycle replica with a clock as the front wheel.
train2017/000000322141.jpg : A room with blue walls and a white sink and door.
train2017/00000016977.jpg : A car that seems to be parked illegally behind a legally parked car
train2017/000000106140.jpg : A large passenger airplane flying through the air.
train2017/000000571635.jpg : A bathroom with a toilet, sink, and shower.

_od-captionsonly.txt_ (<image_caption>)
> A bicycle replica with a clock as the front wheel.
A room with blue walls and a white sink and door.
A car that seems to be parked illegally behind a legally parked car
A large passenger airplane flying through the air.
A bathroom with a toilet, sink, and shower.

Here is an example image and caption:
![image](https://user-images.githubusercontent.com/32551374/104190487-76918d80-541c-11eb-9b8c-a312475100d1.png)
_The man at bat readies to swing at the pitch while the umpire looks on._

I have written this hoping it will be somewhat compatible with @htoyryla's code. I think it should work out-of-the-box. @htoyryla's training script is also included in the Colab. Feel free to use this to generate a dataset for DALL-E and/or to continue on @htoyryla's work.

**Comment by htoyryla** [comment]
Looks good. If using with my trainDALLE, just check that the whole path is reconstructed correctly at this line https://github.com/htoyryla/DALLE-pytorch/blob/main/trainDALLE.py#L171

I probably will not start training with COCO myself, cannot tie up my resources for this. 

Also, I have used a fixed learning rate for these initial experiments, but one should consider adding a an lr scheduler.

**Comment by lucidrains** [comment]
@mrconter1 Thanks for this! I'll start thinking about how I can setup an automatic command-line tool training based off COCO!

**Comment by htoyryla** [comment]
This might work for the tokenizer:

```
import spacy
spacy_en = spacy.load('en_core_web_sm')

def tokenizer(text): 
    return [tok.text for tok in spacy_en.tokenizer(text)]
```

In fact I was planning to use spacy but as my captions were so simple used the plain split(' ') instead.

**Comment by lucidrains** [comment]
> This should work for the tokenizer:
> 
> ```
> import spacy
> spacy_en = spacy.load('en_core_web_sm')
> 
> def tokenizer(text): 
>     return [tok.text for tok in spacy_en.tokenizer(text)]
> ```
> 
> In fact I was planning to use spacy but as my captions were so simple used the plain split(' ') instead.
> 
> In addition to installing spacy (via conda or pip) you need to install the language model:
> 
> ```
> python -m spacy download en_core_web_sm
> ```

nice! i was going to use huggingface's tokenizer, or youtokentome, but spacy looks good too!

**Comment by mrconter1** [comment]
No problem, @lucidrains! I'm not an expert on ML but I am okay at writing scripts. Just ask if you want me to add something or make it more usable. I'm happy to help!

**Comment by Erdene-Ochir0417** [comment]
Hello Thanks for sharing your work
I have looked your code and I have found bug.
I think you should change image_id = int(info['id']) to image_id = int(info['image_id']). Picture does not fit to caption.
Am I wrong?
